app/models/shard_config.rb

```diff
@@ -39,7 +39,10 @@ def self.name_parent_rank_restriction?
     name_parent_rank_restriction
   end
 
-  def self.shard_group_name?
-      ShardConfig.find_by(name: "shard group name").value
-    end
+  def self.shard_group_name
+    results = ShardConfig.find_by(name: "shard group name")
+    return "NSL" if results.nil?
+    results.value
+
+  end
 end
```

Comment:
I like that you are allowing for missing values - something for us to look at more generally in ShardConfig, to make new installation easier to troubleshoot.  

Possible ruby-idiom:  ShardConfig.find_by(name: 'menu label').try('value') || 'NSL'

It occurs to me we could have a private method in ShardConfig to apply this approach easily to every look up for a config value.

Reply:
Changed this to use the ruby-idiom. Ran unit tests on models and controllers. Zero errors